types/dummy/index.d.ts

```diff
@@ -3,3 +3,15 @@
  * ember-data. They will not be shipped to consumers. Ember-data still relies
  * on some private Ember APIs -- those should be defined here as we encounter them.
  */
+
```

Comment:
these do not exist in the ember.js repo / or the mixin system is throwing TS off (which I think is the case for detect?)

Reply:
I don't see a `detect` function being exported from `@ember/array` either in [the API docs](https://emberjs.com/api/ember/release/modules/@ember%2Farray) or in the [RFC 176 data](https://github.com/ember-cli/ember-rfc176-data); does importing that actually work?

Re: `@ember/ordered-set`, that's [a standalone package](https://github.com/emberjs/ember-ordered-set), not a part of core Ember.

Reply:
FWIW, `detect` is on `Mixin` [see here](https://github.com/emberjs/ember.js/blob/19f5c8919052d1acea9147b341dbbc8009c34f59/packages/%40ember/-internals/metal/lib/mixin.ts#L606) (and therefore `Ember.Array` mixin does have a `.detect` method).  Not sure if that was the specific question though...

Reply:
Right, but this declaration is saying you should be able to `import { detect } from '@ember/array';` (which afaik you can't, since `detect` is a method on the mixin, not a module export, right? 🙂)

Reply:
Yep, I totally missed that this was an import. I'll put my history lessons away, sorry for the noise 😸